From fbf8563dea38a8e209d3289ddcab4b18760842a8 Mon Sep 17 00:00:00 2001
From: Misty De Meo <mistydemeo@github.com>
Date: Wed, 15 Jun 2016 15:02:13 -0700
Subject: [PATCH] Fix GetTime.c on Darwin with clock_gettime

On Darwin versions with clock_gettime, #ifdefs will prevent the
mach-specific time functions from being used in most places, and
the mach time headers won't be included; however, this section
was guarded incorrectly and would still try to use them.
---
 rts/posix/GetTime.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rts/posix/GetTime.c b/rts/posix/GetTime.c
index 130d3f1..4d25795 100644
--- a/rts/posix/GetTime.c
+++ b/rts/posix/GetTime.c
@@ -29,14 +29,14 @@
 // we'll implement getProcessCPUTime() and getProcessElapsedTime()
 // separately, using getrusage() and gettimeofday() respectively
 
-#ifdef darwin_HOST_OS
+#if !defined(HAVE_CLOCK_GETTIME) && defined(darwin_HOST_OS)
 static uint64_t timer_scaling_factor_numer = 0;
 static uint64_t timer_scaling_factor_denom = 0;
 #endif
 
 void initializeTimer()
 {
-#ifdef darwin_HOST_OS
+#if !defined(HAVE_CLOCK_GETTIME) && defined(darwin_HOST_OS)
     mach_timebase_info_data_t info;
     (void) mach_timebase_info(&info);
     timer_scaling_factor_numer = (uint64_t)info.numer;
-- 
2.9.0

